packages/lodestar/src/network/reqresp/encoders/responseEncode.ts

```diff
@@ -26,20 +27,23 @@ import {
 export function responseEncodeSuccess(
   config: IBeaconConfig,
   protocol: Protocol
-): (source: AsyncIterable<ResponseBody>) => AsyncIterable<Buffer> {
+): (source: AsyncIterable<LodestarResponseBody>) => AsyncIterable<Buffer> {
   const contextBytesType = contextBytesTypeByProtocol(protocol);
 
   return async function* responseEncodeSuccessTransform(source) {
     for await (const chunk of source) {
       // <result>
       yield Buffer.from([RespStatus.SUCCESS]);
 
-      // <context-bytes>
+      // <context-bytes> - from altair
       const forkName = getForkNameFromResponseBody(config, protocol, chunk);
       yield* writeContextBytes(config, contextBytesType, forkName);
 
       // <encoding-dependent-header> | <encoded-payload>
-      const type = getResponseSzzTypeByMethod(protocol, forkName);
+      const type =
+        protocol.method === Method.BeaconBlocksByRange || protocol.method === Method.BeaconBlocksByRoot
+          ? null
+          : getResponseSzzTypeByMethod(protocol, forkName);
```

Comment:
Adding this information here breaks the abstracting layer of the encoders decoders. I would propose to move this info inside `getResponseSzzTypeByMethod`, and here do
```ts
getResponseSzzTypeByMethod(protocol, forkName, Direction.Outgoing)
```

Reply:
Modifying `getResponseSzzTypeByMethod()` would escalate the change to `responseDecode` and other type definition that get return type from this method so I implement the change to a new method `getOutgoingSerializerByMethod()`